### DISCUSSION TITLE
setting nic's speed before bonding on debian server

### DISCUSSION THREAD

**Discussion by jorigel** [ltsp/ltsp#391] [Q&A]
Hi,
my debian server initializes its nics with 100Mb/sec with auto-negotiation. Using only one nic and setting the speed with e.g. :

`ethtool -s eth0 1000 duplex full autoneg off
`

makes it working with noticeable  higher speed (says 1000 Mb/sec). Maybe its less but noticeable quicker than before.

On the other hand I can set up a bonding with the two nics in network/interfaces like:

```
# The bond interface

auto bond0
iface bond0 inet static
    address   10.1.1.204
    netmask   255.255.255.0
    network   10.1.1.0
    broadcast 10.1.1.255
    gateway   10.1.1.200
    # dns-* options are implemented by the resolvconf package, if installed
    dns-search medianet
    dns-nameserver 10.1.1.230
        bond-slaves eth0 eth1
        bond-primary eth1
        bond-mode 6
        bond-miimon 100
        bond-downdelay 400
        bond-updelay 800
```

and that's also quicker than only one nic telling:

```
ethtool bond0 ->

Settings for bond0:
	Supported ports: [ ]
	Supported link modes:   Not reported
	Supported pause frame use: No
	Supports auto-negotiation: No
	Supported FEC modes: Not reported
	Advertised link modes:  Not reported
	Advertised pause frame use: No
	Advertised auto-negotiation: No
	Advertised FEC modes: Not reported
	Speed: 200Mb/s
	Duplex: Full
	Port: Other
	PHYAD: 0
	Transceiver: internal
	Auto-negotiation: off
	Link detected: yes
```

So gross double the speed of one nic and noticeable quicker than only one nic at work.

Unfortunately I don't get managed to set the speed of the two nics to 1000Mb/sec each and then bundle them to a bonding to get 2000Mb/sec on the server to serve the clients.

Has anybody an idea how to manage this construct?

regards
jori

**Comment by alkisg** [comment]
Jori, all the times when I've had auto-negotiation issues, the problem was in cabling.
Try to replace the LAN cables. Try to connect the server directly to the switch. Don't specify any autonegotiation settings. Temporarily you could also try removing the bond. Check the autonegotiation results. Measure the effective network speed with the [epoptes](https://epoptes.org) LAN benchmark.

If all these don't work, maybe you should ask in a networking list. They'll have more experience with such things, and they're not really related to LTSP.

**Reply by jorigel** [reply]
Hi Alkis,
the server has been hooked up to different switches/different ports directly, even without additional clients, the cable was self-manufactured but meanwhile changed against industrial produced CAT6 patch-cable. All without any changes in result.

Ok, thank You for trying to help and Your personal attention. I'm going to look for a forum for networking affairs getting closer to the problem.
thank U
jori
